db/keyval/etcdv3/mocks/embeded_etcd.go

```diff
@@ -33,6 +34,10 @@ func (embd *Embedded) Start(t *testing.T) {
 
 	cfg := embed.NewConfig()
 	cfg.Dir = dir
+	lpurl, _ := url.Parse("http://localhost:0")
```

Comment:
PR says `fix some issues with tests`. This is one of the issues I often run into when running ETCD on default port, tests fail because the port is already used, but using port 0, some free port will be automatically assigned by system and can run tests with embedded ETCD server.